plugins/techdocs-backend/src/DocsBuilder/builder.ts

```diff
@@ -144,6 +144,18 @@ export class DocsBuilder {
       }
     }
 
+    // Cache downloaded source files for 30 minutes.
+    // TODO: When urlReader/readTree supports some way to get latest commit timestamp,
+    // it should be used to invalidate cache.
+    if (type === 'url') {
```

Comment:
does this mean, if I update my documentation within these 30 minutes, it will not rebuild my docs and display the new changes? 🤔

Reply:
if using the urlReader only of course

Reply:
Yes, as of now. Not great experience. But still the benefit of initial load makes it great. And these two features will improve it -
1. https://github.com/backstage/backstage/issues/3717
2. Caching using timestamp of the commit. (I guess we'll have this one soon).

Reply:
I see, maybe could be a good idea to just mention that in the documentation of how-to-use? So when you use it, you are aware of this side effect?

Reply:
Good idea! Will add in https://github.com/backstage/backstage/pull/4062 since it adds the docs. Thanks! :)

Reply:
Sorry @OrkoHunter for convo on a closed PR, but this code is fresh enough I think...

I'm finding that if I have an entity in the catalog set up for TechDocs, but the entity itself is included in the catalog as a file (rather than a URL), TechDocs will rebuild the entire docs site on every single page load.

Is that a common enough use-cases that we should support caching there?

Reply:
@iamEAP Just be clear about the setup -

We add a file based entity in our `app-config.local.yaml`
```yaml
catalog:
  locations:
    ...
    - type: file
      target: /Users/eric/some/local/entity/catalog-info.yaml
```

And in the `catalog-info.yaml` we have

```yaml
metdata:
  annotations:
    backstage.io/techdocs-ref: "dir:./"
```

Or did you mean some other setup? I am guessing this will ideally only be done in a local development setup by a "Backstage contributor" to TechDocs. A rare use case indeed to implement caching.

The `file` way of defining the catalog location makes it challenging to figure out how to implement the caching. (The directory may not be a git repository. Doing a checksum is an option.)

--- 

On a related note, could you also look also look at https://github.com/backstage/backstage/issues/3717? 🙏 I think the async build will be a nice improvement in the current user experience. I find the current way of "build or check if site is outdated before serving any request (either a page, sub-page, css, _any request to backend_)" architecture very unusual, and not scalable at all. Imagine a high traffic Backstage where TechDocs is exhausting GitHub API rate limit because it checks if the site is outdated on every user request to TechDocs. _sweats_

Reply:
@OrkoHunter Yup, you've got the use-case correct.  Checksum would be an option, or perhaps even just directory/file modified date?  Just to get the simple time-based caching in place.

I like the async idea, though.  Will follow.

Reply:
Good ideas. Will create an issue about it! 💯